clients/da-vinci-client/src/main/java/com/linkedin/davinci/consumer/VeniceChangelogConsumerImpl.java

```diff
@@ -172,48 +172,52 @@ public CompletableFuture<Void> internalSubscribe(Set<Integer> partitions, String
+          if (partitions.contains(topicPartition.getPartitionNumber())) {
+            pubSubConsumer.unSubscribe(topicPartition);
+          }
+        }
+
+        List<PubSubTopicPartition> topicPartitionList =
             getPartitionListToSubscribe(partitions, topicPartitionSet, topicToSubscribe);
-        List<TopicPartition> topicPartitionListToSeek =
+        List<PubSubTopicPartition> topicPartitionListToSeek =
             getPartitionListToSubscribe(partitions, Collections.EMPTY_SET, topicToSubscribe);
 
-        topicPartitionSet.addAll(topicPartitionList);
-        kafkaConsumer.assign(topicPartitionSet);
-        for (TopicPartition topicPartition: topicPartitionList) {
-          if (!topicPartition.topic().endsWith(ChangeCaptureView.CHANGE_CAPTURE_TOPIC_SUFFIX)) {
-            compressorMap.put(topicPartition.partition(), getVersionCompressor(topicPartition));
+        for (PubSubTopicPartition topicPartition: topicPartitionList) {
+          if (!topicPartition.getPubSubTopic().getName().endsWith(ChangeCaptureView.CHANGE_CAPTURE_TOPIC_SUFFIX)) {
+            compressorMap.put(topicPartition.getPartitionNumber(), getVersionCompressor(topicPartition));
           }
         }
-        kafkaConsumer.seekToBeginning(topicPartitionListToSeek);
+        for (PubSubTopicPartition topicPartition: topicPartitionListToSeek) {
+          pubSubConsumer.subscribe(topicPartition, OffsetRecord.LOWEST_OFFSET);
+        }
         return null;
       }
     });
   }
 
-  protected VeniceCompressor getVersionCompressor(TopicPartition topicPartition) {
+  protected VeniceCompressor getVersionCompressor(PubSubTopicPartition topicPartition) {
     Store store = storeRepository.getStore(storeName);
-    Version version = store.getVersion(Version.parseVersionFromVersionTopicName(topicPartition.topic())).get();
+    String topicName = topicPartition.getPubSubTopic().getName();
+    Version version = store.getVersion(Version.parseVersionFromVersionTopicName(topicName)).get();
     VeniceCompressor compressor;
     if (CompressionStrategy.ZSTD_WITH_DICT.equals(version.getCompressionStrategy())) {
-      compressor = compressorFactory.getVersionSpecificCompressor(topicPartition.topic());
+      compressor = compressorFactory.getVersionSpecificCompressor(topicName);
       if (compressor == null) {
         // we need to retrieve the dictionary from the kafka topic
-        ByteBuffer dictionary = DictionaryUtils.readDictionaryFromKafka(
-            topicPartition.topic(),
-            new VeniceProperties(changelogClientConfig.getConsumerProperties()));
-        compressor = compressorFactory.createVersionSpecificCompressorIfNotExist(
-            version.getCompressionStrategy(),
-            topicPartition.topic(),
-            dictionary.array());
+        ByteBuffer dictionary = DictionaryUtils
+            .readDictionaryFromKafka(topicName, new VeniceProperties(changelogClientConfig.getConsumerProperties()));
```

Comment:
Do we have a plan for static functions like this?  Maybe need not be done in this PR, but generally curious.

Reply:
You mean replace the String topic to `PubSubTopic`?

Reply:
That'd probably be the ideal.  readDictionaryFromKafka is static and makes the assumption that we're working with kafka topics.

Reply:
I checked, seems this `readDictionaryFromKafka` has been called in many places like VPJ, integration tests and here. I think it might be better to clean up this in the future.